File: java-solution/src/main/java/com/thiefspin/javasolution/entity/Country.java
package com.thiefspin.javasolution.entity;

import lombok.Getter;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Getter
@Entity(name = "countries")
public class Country {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String currencyCode;
    private Double currencyValue;

}

File: scala-solution/src/main/scala/com/thiefspin/model/Country.scala
package com.thiefspin.model

import anorm.{Macro, RowParser}

final case class Country(
  id: Long,
  name: String,
  currencyCode: String,
  currencyValue: Double
)

object Country {
  val parser: RowParser[Country] = Macro.indexedParser[Country]
}
